src/librustc_driver/driver.rs

```diff
@@ -934,6 +934,7 @@ pub fn phase_3_run_analysis_passes<'tcx, F, R>(sess: &'tcx Session,
     passes.push_pass(MIR_CONST, mir::transform::rustc_peek::SanityCheck);
 
     // What we need to run borrowck etc.
+    passes.push_pass(MIR_CONST, mir::transform::borrow_check::BorrowckMir);
     passes.push_pass(MIR_VALIDATED, mir::transform::qualify_consts::QualifyAndPromoteConstants);
     passes.push_pass(MIR_VALIDATED,
```

Comment:
I see a `SimplifyBranches` pass here within the `MIR_VALIDATED` pipeline. It's important that this won't run *before* borrowck, because otherwise code within an `if false` won't be checked.